oap-server/server-cluster-plugin/cluster-zookeeper-plugin/src/main/java/org/apache/skywalking/oap/server/cluster/plugin/zookeeper/ServiceCacheManager.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.cluster.plugin.zookeeper;
+
+import java.util.*;
+import org.apache.curator.x.discovery.ServiceCache;
+import org.apache.skywalking.oap.server.core.cluster.InstanceDetails;
+
+/**
+ * @author peng-yongsheng
+ */
+public class ServiceCacheManager {
+
+    private final Map<String, ServiceCache<InstanceDetails>> serviceCacheMap;
+
+    public ServiceCacheManager() {
+        this.serviceCacheMap = new HashMap<>();
```

Comment:
Are you sure about the OP on `put` is synchronized already?  No concurrency?

Reply:
I just can know there should be multiple service names in this Map, but do they just be added during startup?